### PR TITLE
New auth instructions for Docker private registry

### PR DESCRIPTION
**Pull request by sascala**

### CODE REVIEW

File: docs/docs/native-docker-private-registry.md [hunk 1]
```diff
@@ -4,34 +4,176 @@ title: Using a Private Docker Registry
...
 # Using a Private Docker Registry
 
-This document describes how to initiate a [Docker](https://docker.com/) pull from
-an authenticated private registry.
+As of Marathon 1.5, you can upload your private Docker registry credentials to a secret store, then reference them in your app or pod definition. This functionality is only available if you are using the Mesos containerizer. If you are using the Docker containerizer, follow [these instructions](#docker-containerizer) to use a private Docker registry.
 
-## Registry  1.0 - Docker pre 1.6 
+## Step 1: Create a Credentials File
+
+1. Log in to your private registry manually. This will create a `.docker` folder and a `.docker/config.json` file.
```
Comment (ichernetsky) [inline docs/docs/native-docker-private-registry.md]: I think it makes sense to not omit `~` in paths related to `config.json`. So, `a .docker folder and a .docker/config.json file` can be substituted with just `a ~/.docker/config.json` file. Also, in the world of filesystems, folders are called directories.

File: docs/docs/native-docker-private-registry.md [hunk 2]
```diff
@@ -4,34 +4,176 @@ title: Using a Private Docker Registry
...
+  ```bash
+  $ docker login some.docker.host.com
+  Username: foo
+  Password:
+  Email: foo@bar.com
+  ```
+
+1. Check that you have the `.docker/config.json` file.
```
Comment (ichernetsky) [inline docs/docs/native-docker-private-registry.md]: ditto

File: docs/docs/native-docker-private-registry.md [hunk 3]
```diff
@@ -4,34 +4,176 @@ title: Using a Private Docker Registry
...
+
+    ```bash
+    $ ls .docker
```
Comment (ichernetsky) [inline docs/docs/native-docker-private-registry.md]: ditto

File: docs/docs/native-docker-private-registry.md [hunk 4]
```diff
@@ -4,34 +4,176 @@ title: Using a Private Docker Registry
...
+    config.json
+    ```
+
+    Your `config.json` file should look like this.
```
Comment (ichernetsky) [inline docs/docs/native-docker-private-registry.md]: ditto

File: docs/docs/native-docker-private-registry.md [hunk 5]
```diff
@@ -4,34 +4,176 @@ title: Using a Private Docker Registry
...
+
+### Step 2: Add the Secret to your App or Pod Definition
+
+#### For an Application
+
+Add the following two parameters to your app definition.
+
+1. A location for the secret in the `secrets` parameter:
+    ```json
+    "secrets": {
```
Comment (ichernetsky) [inline docs/docs/native-docker-private-registry.md]: The indentation is inconsistent.

File: docs/docs/native-docker-private-registry.md [hunk 6]
```diff
@@ -4,34 +4,176 @@ title: Using a Private Docker Registry
...
+          "pullConfigSecret": {
+              "source": "/mesos-docker/pullConfig"
+          }
+      }
+    ```
+
+1. A reference to the secret in the `docker.pullConfig` parameter:
+    ```json
+    "docker": {
```
Comment (ichernetsky) [inline docs/docs/native-docker-private-registry.md]: ditto

File: docs/docs/native-docker-private-registry.md [hunk 7]
```diff
@@ -4,34 +4,176 @@ title: Using a Private Docker Registry
...
+              }
+    }
+    ```
+
+  **Note:** This functionality is _only_ supported with the Mesos containerizer: `container.type` must be `MESOS`.
+
+1. A complete example:
+
+      ```json
+      {
```
Comment (ichernetsky) [inline docs/docs/native-docker-private-registry.md]: The indentation is consistent within the example, and not consistent throughout the document. 2 vs 4 vs N space characters.

File: docs/docs/native-docker-private-registry.md [hunk 8]
```diff
@@ -4,34 +4,176 @@ title: Using a Private Docker Registry
...
+
+1. The Docker image will now pull using the provided security credentials given.
+
+#### For a Pod
+
+Add the following two parameters to your pod definition.
+
+1. A location for the secret in the `secrets` parameter:
+    ```json
+    "secrets": {
```
Comment (ichernetsky) [inline docs/docs/native-docker-private-registry.md]: ditto

File: docs/docs/native-docker-private-registry.md [hunk 9]
```diff
@@ -4,34 +4,176 @@ title: Using a Private Docker Registry
...
+
+  ```
+  "containers": [
+    {
+      "image": {
+        "pullConfig": {
+          "secret": "pullConfigSecret"
+        },
+        "kind": "DOCKER"
+  ]
```
Comment (ichernetsky) [inline docs/docs/native-docker-private-registry.md]: Missing `}`

File: docs/docs/native-docker-private-registry.md [hunk 10]
```diff
@@ -4,74 +4,221 @@ title: Using a Private Docker Registry
...
+          "pullConfig": {
+            "secret": "pullConfigSecret"
+          },
+          "kind": "DOCKER"
+        }
+      }
+    ]
+    ```
+
+    **Note:** This functionality is only supported if `containers.kind` is set to `DOCKER`.
```
Comment (jdef) [inline docs/docs/native-docker-private-registry.md]: s/`containers.kind`/`image.kind`/

**Pull request merged**